### ISSUE TITLE
[request] Support “Save to Nextcloud” direct action for WebKit applications

### ISSUE THREAD

**Issue by fiveangle** [nextcloud/ios#1833]
Thanks for continued work on this project !

The lack of direct saving to Nextcloud is the most-missed feature I have after moving from Dropbox. It would be wonderful if Nextcloud supported this interface directly for all WebKit applications but specifically Safari.

Currently, I have to "save to files" then go to nextcloud then select "upload file" then select the locally-downloaded file, then go back into the files.app and delete the locally-downloaded file.

As an example, observe how Dropbox is presented here from the Safari Actions menu:

![image](https://user-images.githubusercontent.com/2219447/147364179-f7db9dd2-9bd6-4c93-8cb4-11ce51e265bf.jpeg)

![image](https://user-images.githubusercontent.com/2219447/147364227-de677441-93e1-4172-a60f-01672e14d748.jpeg)

Thanks !

-=dave

**Comment by thisIsTheFoxe** [comment]
Thanks for your feedback. 
However, please note that the app already comes with a Share extension.
Are you unable to "Save to Nextcloud" using that feature? If so, please elaborate. What would you expect to happen when triggering the Action? (How) Would it be different from the current Share extension?

**Comment by fiveangle** [comment]
See above screenshot.  There is no such ability to save directly to Nextcloud. 

Expectation is that Nextcloud would register as a "Other action" within iOS so that the user can select Nextcloud as a target app for the share extension with "Save to Nextcloud" just as Dropbox does in the above screenshot.

**Comment by thisIsTheFoxe** [comment]
Thanks for clarifying. 
Please also note that like the Actions you can also edit your Share suggeations in the "More" view you mentioned.

If I understand correctly your request doesn't include any different functionality from the Share extension, right? My concerns with that would be that one would have two ways of doing the exact same thing in the same UX 'group'. This could be confusing and overwhelming for users. (Which is the *real* one? Which one should I use?)

**Comment by ncheng89** [comment]
I encountered a problem that in the media on the mobile phone, long press the picture and click to view in the folder, and the picture preview cannot be opened. I am not sure whether it is a software problem or an ios system problem, but currently my three devices have this problem , The devices are (ipad pro, iphone13 pro, iphone7 plus, ios versions are the latest), please troubleshoot the problem, the service version of nextcloud is 23

**Comment by thisIsTheFoxe** [comment]
@ncheng89 I am sorry, but I am unable to see how your problem relates to this issue. If it does, please explain. If it does not, please open a new issue. 
Thanks.

**Comment by ncheng89** [comment]
Sorry, it is not related to  @thisIsTheFoxe  question, it is a new question, I will create a new question, thank you for your reminder

**Comment by chrissi5120** [comment]
> 

To be honest, this is AFAIK the default approach for most of all productivity apps, please try to find all the different ways to save a "Microsoft Office lens" -scan.

In my opinion, it is always wise to integrate an app like Nextcloud as deep into the OS as possible because “share” tends to be more confusing.

Nextcloud offers different "operational modes", all of them should be accessible to the UX directly..

- took a screenshot? -> Save it on Nextcloud!
- trying to download a file -> Save it on Nextcloud!
- selecting multiple files in your camera roll? -> collect them and save them on nextcloud!
- ...

I don't see anything bad in having multiple ways to achieve the same thing, this is a core ingredient in modern computing

**Comment by fiveangle** [comment]
@chrissi5120 - I think the fact that Apple themselves promote this type of streamlined method for direct-saving to specific classes of apps instead of the swipe-swipe-swipe-swipe-swipe... tap default illustrates that if Apple knows anything about anything, it's designing systems to be as least "confusing" as is possible.  It's not always possible, but in this instance, Apple is solving both confusion for the layperson, as well as utter frustration for the veteran.  Second-guessing Apple's design is really a decision for them to make. I doubt they came to the design lightly, and I for one find the current Nextcloud default difficult to deal with given the number of apps I have installed.

**Comment by fiveangle** [comment]
> Thanks for clarifying. Please also note that like the Actions you can also edit your Share suggeations in the "More" view you mentioned.

That is wonderful !
> 
> If I understand correctly your request doesn't include any different functionality from the Share extension, right?

I don't know.  As I am not an iOS developer, I do not know what exactly constitutes the "Share extension". I thought my description above was thorough but if there is still doubt, please ask more detailed specifics and I will answer explicitely.

> My concerns with that would be that one would have two ways of doing the exact same thing in the same UX 'group'. This could be confusing and overwhelming for users. (Which is the _real_ one? Which one should I use?)


Anyone who uses iOS knows how this interface functions. No one using iOS believes an icon or line is representative of a file system entity.  In fact, only in the last several years has iOS even supported a psuedo "filesystem" in the form of the "Files" app and that was only out of desperation to entice users to consolidate all their file storage into iCloud.  A byproduct of that work is that Nextcloud reaps the same benefits.

**Comment by fiveangle** [comment]
2 years later and we still have to jump through hoops to save files into Nextcloud from many apps. Most often, I "Share..." to the Files app to save locally on the device, then open the Files app to "Save..." into Nextcloud.

**Comment by yannpub** [comment]
Just installed NextCloud, and how to 'save to Nextcloud' was my first search. Looks like a basic UX functionality, very surprised it isn't available.

**Comment by fiveangle** [comment]
> Just installed NextCloud, and how to 'save to Nextcloud' was my first search. Looks like a basic UX functionality, very surprised it isn't available.

Yes, I too am surprised by the resistance to implement the feature
